lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -628,8 +628,9 @@ static void optimizeTranspose(Function *F) {
       continue;
     }
     // Create a new variable NV to hold the transposed result.
-    auto *NV = F->getParent()->createVariable(
-        TN->getType(), V->getName(), V->getVisibilityKind(), V->getTrainKind());
+    auto *NV = F->getParent()->createVariable(TN->getType(), V->getName(),
```

Comment:
This looks good, but it seems that it does not matter as the payload of NV is re-initialized right after that with transposed values of V payload (genericTranspose(&V->getPayload(), &NV->getPayload(), TN->getShuffle())).
So initial value will not matter. But certainly, fixed version is better.